plugins/plugin-docker/che-plugin-docker-compose/src/main/java/org/eclipse/che/plugin/docker/compose/yaml/ComposeEnvironmentParser.java

```diff
@@ -0,0 +1,173 @@
+    /**
+     * Parses compose file from {@link Environment} into {@link CheServicesEnvironmentImpl}.
+     *
+     * @param environment
+     *         environment with {@link EnvironmentRecipe} to parse.
+     *         {@link EnvironmentRecipe} contains {@link CheServicesEnvironmentImpl} definition.
+     * @throws IllegalArgumentException
+     *         when environment or environment recipe is invalid
+     * @throws ServerException
+     *         when environment recipe can not be retrieved
+     */
+    @Override
+    public CheServicesEnvironmentImpl parse(Environment environment) throws IllegalArgumentException,
+                                                                            ServerException {
+        EnvironmentRecipe recipe = environment.getRecipe();
+        String content = getContentOfRecipe(recipe);
+        ComposeEnvironment composeEnvironment = parse(content, recipe.getContentType());
+        return asCheEnvironment(composeEnvironment);
+    }
+
+    /**
+     * Parses compose file into Docker Compose model.
+     *
+     * @param recipeContent compose file to parse
+     * @throws IllegalArgumentException
+     *         when environment or environment recipe is invalid
+     * @throws ServerException
+     *         when environment recipe can not be retrieved
+     */
+    public ComposeEnvironment parse(String recipeContent, String contentType) throws IllegalArgumentException,
+                                                                                         ServerException {
+        ComposeEnvironment composeEnvironment;
+        switch (contentType) {
+            case "application/x-yaml":
+            case "text/yaml":
+            case "text/x-yaml":
+                try {
+                    composeEnvironment = YAML_PARSER.readValue(recipeContent, ComposeEnvironment.class);
+                } catch (IOException e) {
+                    throw new IllegalArgumentException(
+                            "Parsing of environment configuration failed. " + e.getLocalizedMessage());
+                }
+                break;
+            default:
+                throw new IllegalArgumentException("Provided environment recipe content type '" +
+                                                   contentType +
+                                                   "' is unsupported. Supported values are: " +
+                                                   "application/x-yaml, text/yaml, text/x-yaml");
+        }
+        return composeEnvironment;
+    }
+    /**
+     * Converts Docker Compose environment model into YAML file.
+     *
+     * @param composeEnvironment Docker Compose environment model file
+     * @throws IllegalArgumentException
+     *         when argument is null or conversion to YAML fails
+     */
+    public String toYaml(ComposeEnvironment composeEnvironment) throws IllegalArgumentException {
+        checkNotNull(composeEnvironment, "Compose environment should not be null");
```

Comment:
1. NPE should be thrown when argument is null
2. Why don't you just use `Objects` or `Preconditions` methods for such checks?
3. Don't list runtime exceptions like `IllegalArgumetnException` in the `throws` list, but still document them